src/nodes/constraint_aware_append/constraint_aware_append.c

```diff
@@ -441,6 +444,9 @@ constraint_aware_append_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPa
 			case T_WorkTableScan:
 			case T_ForeignScan:
 			case T_CustomScan:
+#if PG14_GE
+			case T_TidRangeScan:
+#endif
 			{
 				List *chunk_clauses = NIL;
 				ListCell *lc;
```

Comment:
Normal code coverage will not discover if this code path has been reached. Have you checked that your test case actually hit this code path? For example, by attaching a debugger and running the test below?